pcdscalc/be_lens_calcs.py

```diff
@@ -925,15 +941,422 @@ def find_z_pos(energy, lens_set, spot_size_fwhm, material=None,
     waist = lam / np.pi * focal_length / w_unfocused
     rayleigh_range = np.pi * waist ** 2 / lam
 
-    logger.info("waist          : %.3e" % waist)
-    logger.info("waist FWHM     : %.3e" % (waist
-                                           * FWHM_SIGMA_CONVERSION / 2.0))
-    logger.info("rayleigh_range : %.3e" % rayleigh_range)
-    logger.info("focal length   : %.3e" % focal_length)
+    logger.info("waist          : %.3e", waist)
+    logger.info("waist FWHM     : %.3e", waist * FWHM_SIGMA_CONVERSION / 2.0)
+    logger.info("rayleigh_range : %.3e", rayleigh_range)
+    logger.info("focal length   : %.3e", focal_length)
 
     w = gaussian_fwhm_to_sigma(spot_size_fwhm) * 2
     delta_z = rayleigh_range * np.sqrt((w / waist) ** 2 - 1)
     z1 = focal_length - delta_z
     z2 = focal_length + delta_z
     z_position = (z1, z2)
     return z_position
+
+
+def plan_set(energy, z_offset, z_range, beam_size_unfocused, size_horizontal,
+             size_vertical=None, exclude=[], max_tot_number_of_lenses=25,
```

Comment:
It doesn't matter here, but sometimes you can run into trouble if you pass a mutable object like a list as the default argument to a function. This is because if you modify this list in the function call, the modified list will be used as the default value in the next call of the function, rather than the original list.

Reply:
Ok, thank you for pointing this out. I can make it None and inside the function do something like this: 
`exclude = exclude or []`?

Reply:
Yes, that is the typical way to handle this, using an immutable sentinel like `None`